sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -65,11 +65,19 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
     checkEvaluationWithOptimization(expr, catalystValue, inputRow)
   }
 
+
+  private def getActualDataType(dt: DataType): DataType = dt match {
```

Comment:
I added a similar method to the `UserDefinedType` companion. Which one shall we add?

Reply:
I can use the method you added.